### PR TITLE
Error message improvements

### PR DESCRIPTION
**Pull request by alexr-bq**
### Summary

Improves error messages and integration tests handling thereof.

### Description

Went through common misconfigurations and our list of errors to make common cases more user friendly. Also fixed integration tests to look for specific error types.

### Related Issue

VER-76910

### Additional Reviewers

@jonathanl-bq 
@NerdLogic

### CODE REVIEW

File: connector/src/main/scala/com/vertica/spark/util/error/ErrorHandling.scala [hunk 1]
```diff
@@ -125,7 +118,6 @@ case class InvalidPartition() extends ConnectorError {
   override def getUserMessage: String = ErrorHandling.appendErrors(this.getFullContext, invariantViolation)
 }
 
-// TODO: Remove
```
Comment (jonathanl-bq) [inline connector/src/main/scala/com/vertica/spark/util/error/ErrorHandling.scala:128]: I still want to remove this. It's not correct to consider this an error.

File: connector/src/main/scala/com/vertica/spark/datasource/core/DSConfigSetup.scala [hunk 2]
```diff
@@ -342,7 +342,7 @@ class DSReadConfigSetup(val pipeFactory: VerticaPipeFactoryInterface = VerticaPi
     config match {
       case DistributedFilesystemReadConfig(_, _, _, _, verticaMetadata, _, _, _) =>
         verticaMetadata match {
-          case None => Left(SchemaDiscoveryError(None))
+          case None => Left(SchemaDiscoveryError())
```
Comment (jonathanl-bq) [inline connector/src/main/scala/com/vertica/spark/datasource/core/DSConfigSetup.scala:345]: I'm glad we're getting rid of some of these errors wrapping Options. I think it'd be nice to get rid of all of them and replace them with .context() calls. Not something to take action on right now, but something I'd like to see going forward.

File: examples/kerberos-example/src/main/scala/example/Main.scala [hunk 3]
```diff
@@ -38,20 +38,25 @@ object Main extends App {
...
     val schema = new StructType(Array(StructField("col1", IntegerType)))
 
     val data = Seq.iterate(0,1000)(_ + 1).map(x => Row(x))
     val df = spark.createDataFrame(spark.sparkContext.parallelize(data), schema).coalesce(1)
     val mode = SaveMode.Overwrite
 
-    df.write.format("com.vertica.spark.datasource.VerticaSource").options(opts + ("table" -> tableName)).mode(mode).save()
-    println("KERBEROS DEMO, WROTE TABLE")
+    //df.write.format("com.vertica.spark.datasource.VerticaSource").options(opts + ("table" -> tableName)).mode(mode).save()
+    //println("KERBEROS DEMO, WROTE TABLE")
```
Comment (jonathanl-bq) [inline examples/kerberos-example/src/main/scala/example/Main.scala:49]: Probably should remove this commented out code.

File: functional-tests/build.sbt [hunk 4]
```diff
@@ -40,5 +40,5 @@ assemblyMergeStrategy in assembly := {
 }
 */
 
-
+//unmanagedClasspath in Runtime += new File("/etc/hadoop/conf/")
```
Comment (jonathanl-bq) [inline functional-tests/build.sbt:43]: I'm confused. We removed this before, but now we're bringing it back?

File: functional-tests/src/main/resources/application.conf [hunk 5]
```diff
@@ -7,5 +7,6 @@ functional-tests {
   log=true
   filepath="hdfs://hdfs:8020/data/"
   dirpath="hdfs://hdfs:8020/data/dirtest/"
+
```
Comment (jonathanl-bq) [inline functional-tests/src/main/resources/application.conf:10]: nit: revert

**Review by jonathanl-bq — CHANGES_REQUESTED** [review]

File: connector/src/main/scala/com/vertica/spark/util/error/ErrorHandling.scala (hunk 1 above)
Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/util/error/ErrorHandling.scala:128]: Noted, but I think we can avoid todo comments like this for the release. Can create a ticket if one doesn't exist.

File: functional-tests/build.sbt (hunk 4 above)
Comment (alexr-bq) [reply functional-tests/build.sbt:43]: I changed my mind, too useful to have if we need it. Should look into making it conditional but I haven't gotten to it.

File: examples/kerberos-example/src/main/scala/example/Main.scala (hunk 3 above)
Comment (alexr-bq) [reply examples/kerberos-example/src/main/scala/example/Main.scala:49]: Fixed

File: connector/src/main/scala/com/vertica/spark/util/error/ErrorHandling.scala (hunk 1 above)
Comment (jonathanl-bq) [reply connector/src/main/scala/com/vertica/spark/util/error/ErrorHandling.scala:128]: I'm not sure what's worse in terms of optics. I guess we can take out the TODO, but I wouldn't be surprised if this "error" raises some eyebrows.

**Review by jonathanl-bq — APPROVED** [review]

**Pull request merged**